src/allmydata/storage_client.py

```diff
@@ -292,28 +306,71 @@ def _check_connected_high_water_mark(self):
                 remaining.append( (threshold, d) )
         self._threshold_listeners = remaining
 
-    def _got_announcement(self, key_s, ann):
-        precondition(isinstance(key_s, str), key_s)
-        precondition(key_s.startswith("v0-"), key_s)
-        precondition(ann["service-name"] == "storage", ann["service-name"])
-        server_id = key_s
+    def _should_ignore_announcement(self, server_id, ann):
+        """
+        Determine whether a new storage announcement should be discarded or used
+        to update our collection of storage servers.
+
+        :param bytes server_id: The unique identifier for the storage server
+            which made the announcement.
+
+        :param dict ann: The announcement.
+
+        :return bool: ``True`` if the announcement should be ignored,
+            ``False`` if it should be used to update our local storage server
+            state.
+        """
+        # Let local static configuration always override any announcement for
```

Comment:
This new method seems to drop two relevant checks: "starts with `v0-`" and "is announcing storage-service".

`precondition` looks an awful lot like `assert` but will actually run all the time as far as I can tell and these checks seem worth saving (unless there's good reasons to drop them?)